From fd11433c7ba4d78da765c80b32eca63a8abe436d Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 8 May 2015 00:30:53 -0700
Subject: [PATCH] Check the result of fread()

Fixes build errors on newer ubuntu machines

| SimpleFileParsing.c: In function 'ProcessFile':
| SimpleFileParsing.c:705:3: error: ignoring return value of 'fread',
declared with attribute warn_unused_result [-Werror=unused-result]
|    fread ((VOID *) SourceFile->FileBuffer, SourceFile->FileSize, 1,
SourceFile->Fptr);
|    ^
| cc1: all warnings being treated as errors
| ../Makefiles/footer.makefile:27: recipe for target
'SimpleFileParsing.o' failed

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 BaseTools/Source/C/Common/SimpleFileParsing.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/BaseTools/Source/C/Common/SimpleFileParsing.c b/BaseTools/Source/C/Common/SimpleFileParsing.c
index 877bb6f..d11ac38 100644
--- a/BaseTools/Source/C/Common/SimpleFileParsing.c
+++ b/BaseTools/Source/C/Common/SimpleFileParsing.c
@@ -702,7 +702,11 @@ Returns:
     return STATUS_ERROR;
   }
 
-  fread ((VOID *) SourceFile->FileBuffer, SourceFile->FileSize, 1, SourceFile->Fptr);
+  size_t sz = fread ((VOID *) SourceFile->FileBuffer, SourceFile->FileSize, 1, SourceFile->Fptr);
+  if (sz != SourceFile->FileSize) {
+    Error (NULL, 0, 4001, "Resource: Unable to read data from file", NULL);
+    return STATUS_ERROR;
+  }
   SourceFile->FileBuffer[(SourceFile->FileSize / sizeof (CHAR8 ))] = T_CHAR_NULL;
   //
   // Pre-process the file to replace comments with spaces
-- 
2.1.4

